network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/ConfiguredBusImpl.java

```diff
@@ -77,6 +82,40 @@ public double getAngle() {
         return getResource().getAttributes().getAngle();
     }
 
+    private void setVInCalculatedBus(CalculatedBusAttributes calculatedBusAttributes, double value) {
+        calculatedBusAttributes.setV(value);
+    }
+
+    private void setAngleInCalculatedBus(CalculatedBusAttributes calculatedBusAttributes, double value) {
+        calculatedBusAttributes.setAngle(value);
+    }
+
+    private void updateCalculatedBusAttributes(double newValue,
+                                               String voltageLevelId,
+                                               ObjDoubleConsumer<CalculatedBusAttributes> setValue) {
+        Optional<VoltageLevelImpl> voltageLevelOpt = index.getVoltageLevel(voltageLevelId);
+
+        voltageLevelOpt.ifPresent(voltageLevel -> {
+            List<CalculatedBusAttributes> calculatedBusAttributesList = voltageLevel.getResource()
+                .getAttributes()
+                .getCalculatedBusesForBusView();
+
+            if (CollectionUtils.isNotEmpty(calculatedBusAttributesList)) {
+                List<Integer> calculatedBusesNum = getAllTerminals().stream()
+                    .filter(Terminal::isConnected)
+                    .map(t -> ((CalculatedBus) t.getBusView().getBus()).getCalculatedBusNum()).distinct().toList();
```

Comment:
all terminals in one configured bus always map to the same busviewbus no ?

Reply:
You're right.
We consider now only the first bus in bus view ...